frontend/pages/api/auth/[...nextauth].tsx

```diff
@@ -0,0 +1,100 @@
+import NextAuth, { Session } from "next-auth";
+import CredentialsProvider from "next-auth/providers/credentials";
+import { JWT } from "next-auth/jwt";
+import { JWT_SIGNING_PRIVATE_KEY, REFRESH_TOKEN_ERROR } from "../../../utils/constants";
+import { Credentials, LoginUser, User } from "../../../types/user";
+import { login, refreshToken } from "../../../api/authService";
+import { AccessToken } from "../../../types/token";
+
+async function refreshAccessToken(prevToken: JWT) {
+  try {
+    const data: AccessToken = await refreshToken(prevToken.refreshToken ?? null);
+    return {
+      ...prevToken,
+      accessToken: data.accessToken.token,
+      accessTokenExpires: Date.now() + +data.accessToken.expiresIn * 1000,
```

Comment:
Just to add better readibility

```suggestion
      accessTokenExpires: Date.now() + Number(data.accessToken.expiresIn) * 1000,
```